src/app/app.component.ts

```diff
@@ -69,20 +80,28 @@ export class AppComponent implements OnInit {
         }
       });
     }
-    
+
+
     // Migrate all accounts
     this.accountService.migrate();
 
-    if (this.globalVars.webview || this.globalVars.inTab || this.globalVars.inFrame()) {
-      // We must be running in a webview OR opened with window.open OR in an iframe to initialize
-      this.identityService.initialize().subscribe(res => {
-        this.globalVars.hostname = res.hostname;
-        if (this.globalVars.isFullAccessHostname()) {
-          this.globalVars.accessLevelRequest = AccessLevel.Full;
-        }
 
+    if (this.globalVars.webview || this.globalVars.inTab || this.globalVars.inFrame()) {
```

Comment:
Which one of these is `true` when there's a callback and why? Seems like none of them should be

Reply:
Didn't really have a way to verify. Seems like one of them does since it works for Andy, but I've reorganized the conditions just to make sure.

Reply:
I was also passing `webview=true` to make it work. Would be best if we didn't need to do that, though.

Reply:
@andyboyd, I fixed it now. Also added `?app` param we've talked about.